### PR TITLE
feat(helm): add command to generate documentation

### PR DESCRIPTION
**Pull request by technosophos**
This adds a hidden 'helm doc' command that generates documentation out
of the Helm help text.

Currently it can create:

- man pages for HELM(1)
- Markdown documentation
- bash autocompletions

Those are the three built-ins for Cobra.

The command is hidden so that it doesn't show up to the regular user,
since this is really a specialized task. It has the advantage of making
it trivially easy to install man pages and bash completion at installation time.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/helm/1484)

### CODE REVIEW

File: cmd/helm/helm.go [hunk 1]
```diff
@@ -115,6 +115,7 @@ func newRootCmd(out io.Writer) *cobra.Command {
 		newUpgradeCmd(nil, out),
 		newVerifyCmd(out),
 		newVersionCmd(nil, out),
+		newDocsCmd(out, cmd),
```
Comment (adamreese) [inline cmd/helm/helm.go]: I would put a comment to make clear this is not part of core functionality

**Comment by technosophos** [comment]
DON'T MERGE THIS UNTIL AFTER 2.0.0!

**Comment by rodcloutier** [comment]
Does this support the completion command to generate the completion script on the fly? Something like 
`helm completion --shell bash`. Maybe it `helm docs --type bash` will also work?

It would be nice to have this since we could deploy only the binary without having to provide the completion script.

**Comment by technosophos** [comment]
@rodcloutier Yeah, it's `helm docs --type bash`. Right now, it just uses the built-in completion that Cobra provides.

**Pull request merged**